File: 0771-jewels-and-stones/0771-jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_map<char, int> map;
        int total = 0;

        for (char ch : stones) {
            map[ch]++;
        }

        for (char ch : jewels) {
            if (map.contains(ch)) {
                total += map[ch];
            }
        }

        return total;
    }
};

File: 0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashMap<Character, Integer> map = new HashMap<>();
        int total = 0;

        for(int i=0; i<stones.length(); i++){
            char ch = stones.charAt(i);

            map.put(ch, map.getOrDefault(ch, 0) + 1);
        }

        for(int i=0; i<jewels.length(); i++){
            char ch = jewels.charAt(i);

            if(map.containsKey(ch)){
                total += map.get(ch);
            }
        }

        return total;
    }
}